monkey/infection_monkey/island_api_client/i_island_api_client.py

```diff
@@ -15,19 +15,19 @@ class IIslandAPIClient(ABC):
     """
 
     @abstractmethod
-    def connect(self, island_server: SocketAddress):
+    def login(self, otp: str):
         """
         Connect to the island's API
 
-        :param island_server: The socket address of the API
-        :raises IslandAPIConnectionError: If the client cannot successfully connect to the island
-        :raises IslandAPIRequestError: If an error occurs while attempting to connect to the
-                                       island due to an issue in the request sent from the client
-        :raises IslandAPIRequestFailedError: If an error occurs while attempting to connect to the
-                                             island due to an error on the server
-        :raises IslandAPITimeoutError: If a timeout occurs while attempting to connect to the island
-        :raises IslandAPIError: If an unexpected error occurs while attempting to connect to the
-                                island
+        :param otp: A one-time password used to authenticate with the Island API
+        :raises IslandAPIConnectionError: If the client cannot successfully connect to the Island
```

Comment:
All of these comments are missing AuthenticationError. I'm not sure this is the PR to fix it though

Reply:
It's not really for this PR, but it's a simple fix so I just did it.